agent/reconciler/reconciler_test.go

```diff
@@ -425,6 +617,63 @@ runCmd:
 				}))
 			})
 
+			Context("When use-installer-controller is set", func() {
+				BeforeEach(func() {
+					hostReconciler.UseInstallerController = true
+				})
+
+				It("should return an error if we fail to load the uninstallation script", func() {
+					byoHost.Spec.UninstallationScript = nil
+					Expect(patchHelper.Patch(ctx, byoHost, patch.WithStatusObservedGeneration{})).NotTo(HaveOccurred())
+
+					result, reconcilerErr := hostReconciler.Reconcile(ctx, controllerruntime.Request{
+						NamespacedName: byoHostLookupKey,
+					})
+					Expect(result).To(Equal(controllerruntime.Result{}))
+					Expect(reconcilerErr).To(MatchError("UninstallationScript not found in Byohost " + byoHost.Name))
+				})
+
+				It("should return error if uninstall script execution failed ", func() {
+					uninstallScript = `testcommand`
+					byoHost.Spec.UninstallationScript = &uninstallScript
+					Expect(patchHelper.Patch(ctx, byoHost, patch.WithStatusObservedGeneration{})).NotTo(HaveOccurred())
+
+					result, reconcilerErr := hostReconciler.Reconcile(ctx, controllerruntime.Request{
+						NamespacedName: byoHostLookupKey,
+					})
+					Expect(result).To(Equal(controllerruntime.Result{}))
+					Expect(reconcilerErr).To(HaveOccurred())
+				})
+
+				FIt("should set K8sComponentsInstallationSucceeded to false if uninstall suceeds", func() {
```

Comment:
Remove focus mode as it will skip other tests in this suite.

Reply:
Removed. Thanks